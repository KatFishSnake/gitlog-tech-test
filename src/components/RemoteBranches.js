import { useState, useEffect } from "react";

import { mockRemote } from "../mockData";
import OriginDropdown from "./OriginDropdown";
import NavBranch from "./NavBranch";

const RemoteBranches = () => {
  const [active, setActive] = useState(null);
  const [origins, setOrigins] = useState([]);

  useEffect(() => {
    setOrigins(mockRemote);
  }, []);

  return (
    <div className="flex flex-col mb-4">
      <h4 className="mb-4 text-xs not-italic font-bold leading-3 tracking-normal text-left text-gray-600">
        REMOTE BRANCHES
      </h4>
      <div className="text-xs">
        {origins.map((item) => (
          <OriginDropdown label={item.label} key={item.id}>
            {item.branches.map((inner) => (
              <NavBranch
                key={inner.id}
                label={inner.label}
                active={inner.id === active}
                onClick={() => {
                  setActive(inner.id);
                }}
              />
            ))}
          </OriginDropdown>
        ))}
      </div>
    </div>
  );
};

export default RemoteBranches;
