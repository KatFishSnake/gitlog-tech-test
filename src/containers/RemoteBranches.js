import { useState } from "react";

import useFetch from "../hooks/useFetch";
import { mockRemote } from "../config";
import OriginDropdown from "../components/OriginDropdown";
import NavItem from "../components/NavItem";
import { StyledNavHeader } from "../components/styled";
import Loader from "../components/Loader";

const RemoteBranches = () => {
  const [active, setActive] = useState(null);
  const { loading, data: origins } = useFetch(mockRemote);

  return (
    <div className="flex flex-col">
      <StyledNavHeader className="mb-4 text-xxs not-italic font-bold leading-3 tracking-normal text-left">
        REMOTE BRANCHES
      </StyledNavHeader>
      <div className="text-xs mb-4">
        {loading ? (
          <Loader />
        ) : (
          origins.map((item) => (
            <OriginDropdown key={item.id} label={item.label}>
              {item.branches.map((inner) => (
                <NavItem
                  key={inner.id}
                  label={inner.label}
                  active={inner.id === active}
                  onClick={() => {
                    setActive(inner.id);
                  }}
                />
              ))}
            </OriginDropdown>
          ))
        )}
      </div>
    </div>
  );
};

export default RemoteBranches;
