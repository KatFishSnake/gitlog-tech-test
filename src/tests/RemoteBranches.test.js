import { render } from "@testing-library/react";

import RemoteBranches from "../containers/RemoteBranches";

test("Renders RemoteBranches consistently", () => {
  const { container } = render(<RemoteBranches />);
  expect(container).toMatchSnapshot();
});
