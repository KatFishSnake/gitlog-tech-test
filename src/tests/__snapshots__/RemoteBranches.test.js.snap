// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RemoteBranches consistently 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h4
      class="sc-bdfBwQ iHCwhp mb-4 text-xxs not-italic font-bold leading-3 tracking-normal text-left"
    >
      REMOTE BRANCHES
    </h4>
    <div
      class="text-xs mb-4"
    >
      <span>
        Loading...
      </span>
    </div>
  </div>
</div>
`;
